src/com/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1264,3 +1243,21 @@
   [action-for-unsupported-fn]
   (fail-on-unsupported action-for-unsupported-fn)
   (warn-on-db-deprecation))
+
+(def ^:dynamic *orphaned-path-gc-limit* 200)
+(def ^:dynamic *orphaned-value-gc-limit* 200)
+
+(defn garbage-collect!
+  "Delete any lingering, unassociated data in the database"
+  [db]
+  (time!
+   (:gc metrics)
+   (jdbc/with-transacted-connection db
```

Comment:
I think you changed the semantics of this here. I think that when you use sql/transaction inside another sql/transaction (which is what with-transacted-connection creates) it doesn't actually create a separate tx from a database perspective. Maybe with-connection here is what you want instead?